doc/content/physics/reactive_infiltration.md

```diff
@@ -0,0 +1,371 @@
+\f]
+
+Example input file that defines the product and solid's inner radius
+```
+[Models]
+  [model]
+    type = ProductGeometry
+    solid_fraction = 'state/phi_s'
+    product_fraction = 'state/phi_p'
+    inner_radius = 'state/ri'
+    outer_radius = 'state/ro'
+  []
+[]
+```
+
+### Product reaction rate {#DiffusionLimitedReaction}
+Model object [DiffusionLimitedReaction](#diffusionlimitedreaction)
+
+The reaction rates dictates the creation of the product. Within our assumptions, the growth of the product is controlled by the diffusion of the liquid through the product to react with the solid at the product-solid interface. Considered the same RVE as in [figure above](@ref rve), with \f$c_P\f$ (units of moles) as the concentration of the product in the RVE. Diffusion equation yields
+\f[
+    \dot{c}_P = \nabla \cdot D\nabla c_P
+\f]
+\f$D\f$ is the diffusion coeficient (untis of area per time) of liquid through the product. Integrating through the product's volume and applying Green's theorem,
+\f[
+    \dot{C} = D \dfrac{C_{liquid-product} - C_{product-solid}}{\delta_P}A_{enclosed}
+\f]
+Here,
+\f{align*}
+     &\ \dot{C} = \dot{\alpha}_P V_{RVE} \quad  C_{product-solid} = 0 \quad C_{liquid-product} = \dfrac{\dot{\alpha}_L V_{RVE}}{\varphi_L V_{RVE}} = \dfrac{1}{\Omega_L} \\
+     &\ A_{enclosed} = 2\pi(r_o + r_i)H \quad \delta_P = r_o - r_i
+\f}
+The step function \f$R_L(\alpha_L)\f$, \f$R_S(\alpha_S)\f$ as reactivity is introduced to supressed the production rate when either the solid or the liquid is fully consumed. Finally, in normalized variables, and let the \f$R = l_c\f$ as the characterisitc length scales of the pores features. The product reaction rate is then,
+\f[
+    \dot{\alpha}_P = \dfrac{2}{\Omega_L}\dfrac{D}{l_c^2}\dfrac{\bar{r}_o+\bar{r}_i}{\bar{r}_o-\bar{r}_i} R_L R_S
+\f]
+
+Example input file that defines the product reaction rate
+```
+[Models]
+  [model]
+    type = DiffusionLimitedReaction
+    diffusion_coefficient = 1e-3
+    molar_volume = 1
+    product_inner_radius = 'state/ri'
+    solid_inner_radius = 'state/ro'
+    liquid_reactivity = 'state/R_l'
+    solid_reactivity = 'state/R_s'
+    reaction_rate = 'state/alpha_rate'
+  []
+[]
+```
+
+### Hermite Smooth Step Function {#HermiteSmoothStep}
+Model object [HermiteSmoothStep](#hermitesmoothstep)
+
+A cubic Hermite interpolation after clambing is used as a smooth step function:
+\f{align*}
+    &0 & x_r \le 0 \\
+    R(x) = & 3x_r^2 - 2x_r^3 & 0 \ge x_r \le 1 \\
+    &0 & x_r \ge 1
+\f}
+where
+\f[
+    x_r = \dfrac{x-x_l}{x_u-x_l}
+\f]
+Here, \f$x_l,x_u\f$ denote the lower and upper bound of the smoothen transition regime.
+
+> The Hermite Function is used in the reactive infiltration for the liquid and solid reactivity \f$R_i\f$
+
+Example input file that use the smooth step function
+
+```
+[Models]
+  [model]
+    type = HermiteSmoothStep
+    argument = 'state/foo'
+    value = 'state/bar'
+    lower_bound = '0'
+    upper_bound = '0.05'
+  []
+[]
+```
```

Comment:
I think I still prefer these object descriptions to go into the source files. Like `options.doc() = "......"`.

Reply:
I was convinced to leave them here.